tests/test_providers.py

```diff
@@ -198,3 +198,61 @@ async def test_provider_params_remove_unused_options(mockserver):
         crawler.stats.get_value("scrapy-zyte-api/request_args/productNavigationOptions")
         is None
     )
+
+
+@ensureDeferred
+async def test_provider_extractfrom(mockserver):
+    from typing import Annotated
+
+    @attrs.define
+    class AnnotatedProductPage(BasePage):
+        product: Annotated[Product, ExtractFrom.httpResponseBody]
+        product2: Annotated[Product, ExtractFrom.httpResponseBody]
+
+    class AnnotatedZyteAPISpider(ZyteAPISpider):
+        def parse_(self, response: DummyResponse, page: AnnotatedProductPage):  # type: ignore[override]
```

Comment:
> error: Argument 2 of "parse_" is incompatible with supertype "ZyteAPISpider"; supertype defines the argument type as "ProductPage"  [override]
> note: This violates the Liskov substitution principle
> note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides

I think this is a general problem with scrapy-poet-annotated callbacks, it will also be worse when we release typed Scrapy as `Scrapy.parse` has `Response` but we often use `DummyResponse`.

Reply:
Could you please open an issue for this in scrapy-poet?

Reply:
I will, but I'm not sure what to do with it :)

Reply:
https://github.com/scrapinghub/scrapy-poet/issues/179